src/common/src/cache.rs

```diff
@@ -766,25 +768,33 @@ impl<K: LruKey + Clone, T: LruValue> LruCache<K, T> {
     ) -> Result<Result<CachableEntry<K, T>, E>, RecvError>
     where
         F: FnOnce() -> VC,
```

Comment:
Seems async closure is unnecessary? We only need a future here. cc @Little-Wallace

Reply:
It's necessary. There's some variables that only need to be computed when cache miss. See other parts of the PR for more information.

Reply:
e.g. we need to call `self.meta_path` to compute the meta path ONLY when cache miss. But `&self` is not `'static`, so we can only compute it in this FnOnce closure and outside the `Future + 'static`.